repack.go

```diff
@@ -0,0 +1,107 @@
+package umoci
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/openSUSE/umoci/mutate"
+	"github.com/openSUSE/umoci/oci/casext"
+	"github.com/openSUSE/umoci/oci/layer"
+	"github.com/openSUSE/umoci/pkg/fseval"
+	"github.com/openSUSE/umoci/pkg/mtreefilter"
+	ispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"github.com/vbatts/go-mtree"
+	"golang.org/x/net/context"
+)
+
+// Repack repacks a bundle into an image adding a new layer for the changed
+// data in the bundle.
+func Repack(engineExt casext.Engine, tagName string, bundlePath string, meta Meta, history *ispec.History, maskedPaths []string, refreshBundle bool, mutator *mutate.Mutator) error {
```

Comment:
I really don't like this function signature, but I don't really want to block this PR over this (in my view, the API should be more builder-like than this). If I merge it like this and change it later, would that be more of a pain to you than if we iron out the API now?